solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -233,10 +233,14 @@ public void setConfigMetadata(String configName, Map<String, Object> data) throw
   @Override
   public Map<String, Object> getConfigMetadata(String configName) throws IOException {
     try {
+
+      byte[] bytedata = zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true);
+      if (bytedata == null) return new HashMap<>();
+
       @SuppressWarnings("unchecked")
       Map<String, Object> data =
           (Map<String, Object>)
-              Utils.fromJSON(zkClient.getData(CONFIGS_ZKNODE + "/" + configName, null, null, true));
+              Utils.fromJSON(bytedata);
```

Comment:
Maybe this should instead make `Utils.fromJSON` to be more lenient and allow `null array->empty map`?

Reply:
yea this works better. I'll update this PR with this change and tests.